manager/controllers/app/select_modules.go

```diff
@@ -324,7 +322,14 @@ func match(source, sink *taxonomy.Interface) bool {
 	if source == nil || sink == nil {
 		return false
 	}
-	return source.DataFormat == sink.DataFormat && source.Protocol == sink.Protocol
+	if source.Protocol != sink.Protocol {
+		return false
+	}
+	// an empty DataFormat value is not checked
+	if source.DataFormat != "" && sink.DataFormat != "" && source.DataFormat != sink.DataFormat {
```

Comment:
in the following two cases the return will be `true` 
source.DataFormat  == "" but sink.DataFormat != ""
source.DataFormat  != "" but sink.DataFormat == ""

is it correct ?